.devcontainer/Dockerfile

```diff
@@ -8,16 +8,74 @@ FROM mcr.microsoft.com/vscode/devcontainers/go:0-${VARIANT}
 ARG NODE_VERSION="none"
 RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
 
-# Download the Microsoft repository GPG keys
-RUN wget https://packages.microsoft.com/config/debian/10/packages-microsoft-prod.deb
-# Register the Microsoft repository GPG keys
-RUN  dpkg -i packages-microsoft-prod.deb
+# Download and register the Microsoft repository GPG keys
+RUN wget https://packages.microsoft.com/config/debian/10/packages-microsoft-prod.deb && \
+    dpkg -i packages-microsoft-prod.deb && \
+    rm -f packages-microsoft-prod.deb
 
 # [Optional] Uncomment this section to install additional OS packages.
 RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get install -y --no-install-recommends fish\
-    && apt-get install -y --no-install-recommends powershell
+    && apt-get install -y --no-install-recommends \
+        fish \
+        powershell \
+        tmux \
+    && apt-get clean
 
+# Deploy latest stable oh-my-posh as a starting point:
+RUN pwsh -Command Install-Module oh-my-posh -Scope AllUsers -Force && \
+    chmod 777 -R /usr/local/share/powershell
+# If you recompile the prompt in the devcontainer you may overwrite the executables at:
+# /usr/local/share/powershell/Modules/oh-my-posh/<version>/bin/
+
+# Deploy symlink of binary under /usr/bin/oh-my-posh for convenience access from non-Powershell shells:
+RUN export ARCH=`uname -m`;\
+    if [[ "$ARCH" -eq "aarch64" ]]; then \
+        export POSH_BIN="$(find /usr/local/share/powershell/Modules/oh-my-posh/*/bin/posh-linux-arm64)";\
+    elif [[ "$ARCH" -eq "armv7l" ]]; then \
+        export POSH_BIN="$(find /usr/local/share/powershell/Modules/oh-my-posh/*/bin/posh-linux-arm)";\
+    else \
+        export POSH_BIN="$(find /usr/local/share/powershell/Modules/oh-my-posh/*/bin/posh-linux-amd64)";\
+    fi;\
+    ln -sf $POSH_BIN /usr/bin/oh-my-posh
+# NOTE: devcontainers are Linux-only at this time but when
+# Windows or Darwin is supported someone will need to improve
+# the code logic above.
+
+# Deploy all latest oh-my-posh themes at devcontainer build time:
+ARG USERNAME=vscode
+ARG POSHTHEMES_ROOT=/home/${USERNAME}/.poshthemes
+ENV POSHTHEMES_ROOT=${POSHTHEMES_ROOT}
+RUN mkdir -m=777 -p ${POSHTHEMES_ROOT} && \
```

Comment:
Same for themes, we can make use of `docker-compose` to volume mount the themes inside the container's folder.

Reply:
From your perspective, what would be the benefit vs unzipping latest at build time?

Reply:
For me, the devcontainer is a tool to help developing/testing omp. I'm not sure it's the best thing to set up omp in it and use it directly. I would be better to give automation through tasks, like building and executing omp in a specific shell. Because if omp is configured with the latest version and I work on it, how do I test the version I'm working on?

Reply:
What I would expect, not knowing the details yet (I should really look into this) is the following:
- container gets built with all dependencies to develop
- the relevant parts are volume mounted (see docker-compose), like the PowerShell module (no need to install, always latest code when changing things)
- it builds the latest code live upon start (so we have oh-my-posh working based on the code)
- I can trigger a new build for all shells/ps module whenever I want

Reply:
> What I would expect, not knowing the details yet (I should really look into this) is the following:
> 
> * container gets built with all dependencies to develop
_**ok**_
> * the relevant parts are volume mounted (see docker-compose), like the PowerShell module (no need to install, always latest code when changing things)
**_can symlink from the user profile to the scripts in the folder_**
> * it builds the latest code live upon start (so we have oh-my-posh working based on the code)
**_not sure about this one. Unless you restart the container, you always use th same instance. We can build each time by using `onCreateCommand` in devcontainer.json but it slows down the container start and probably that yo know when you want to build_**  
> * I can trigger a new build for all shells/ps module whenever I want
_**can create a new command in `tasks.json`**_

Reply:
@lnu we could indeed skip the third one as long as you have the task to make it up-to-date. This was me writing what I'd expect without any constraints :-)

Reply:
>> if omp is configured with the latest version and I work on it, how do I test the version I'm working on?

So, the way the PR is at the time I write this, you could technically enter Bash and rebuild over the symlink source path, so something like this should work:

```shell
bash
cd src
go build -o $(readlink /usr/bin/oh-my-posh)
```
(I ran "bash" because it's currently configured to start your shell in Powershell by default.)

Doing so would overwrite your live install and immediately replace your working shell with your own current version.

Could also predeploy some aliases to do this for you at build time, if you think that's a good idea. Adding to tasks.json is an option as well, but remember you can't make tasks appear *only* within the devcontainer, so folks who clone this in VSCode and don't use the remote containers extension may accidentally run the task they weren't supposed to, so we'd probably want to name such a task something obvious like "_devcontainer: rebuild installed_".